nixos/modules/services/ttys/kmscon.nix

```diff
@@ -101,11 +109,20 @@ in {
     systemd.services.systemd-vconsole-setup.enable = false;
     systemd.services.reload-systemd-vconsole-setup.enable = false;
 
-    services.kmscon.extraConfig =
+    services.kmscon.extraConfig = lib.optionalString cfg.useXkbConfig (
+      with config.services.xserver; ''
+        xkb-model=${xkbModel}
+        xkb-layout=${layout}
+        xkb-options=${xkbOptions}
+      '' + lib.optionalString (xkbVariant != "") ''
+        xkb-variant=${xkbVariant}
```

Comment:
Nice! These `xserver` settings were renamed since you created the PR, which enables you to do something like this:

```nix
services.kmscon.extraConfig = lib.optionalString cfg.useXkbConfig (
  lib.mapAttrs
    (n: v: if v != "" then "xkb-${n}=${v}" else "")
    config.services.xserver
) + ( # ...
```

This way, the `!= ""` check would be consistent with all options, you get rid of the `with <>;` prefix, and you rely upon the `xkb.${n}` naming of the options (thanks to the renaming), which is consistent thankfully with the `kmscon` naming of the options.

Reply:
updated per your suggestion